### PR TITLE
Set default pin priority for apt

### PR DESCRIPTION
**Pull request by wyardley**
#### Pull Request (PR) description
It seems as if on some Debian distributions, the upstream package takes priority when the apt repo isn't pinned. Set the priority to 200, since this module configures and uses non-default repos

### CODE REVIEW

**Comment by wyardley** [comment]
I merged #3 on red, since I think #2 and #3 were prerequisites for cutting a release and / or getting tests passing.
I'm not sure if this is the right fix / behavior, so opening this PR for discussion.

File: manifests/init.pp [hunk 1]
```diff
@@ -73,7 +74,7 @@
 class erlang (
   String  $package_name   = 'erlang',
   String  $package_ensure = 'installed',
-  Optional[Variant[Numeric, String]] $package_apt_pin = undef,
+  Optional[Variant[Numeric, String]] $package_apt_pin = 200,
```
Comment (wyardley) [inline manifests/init.pp:77]: If it's still not recommended to have a default value for optional parameters, open to suggestions on the best way to proceed here...

**Comment by wyardley** [comment]
Ubuntu 20 failing setup (with that network error) has been happening consistently, and I think is unrelated.

File: manifests/init.pp (hunk 1 above)
Comment (ekohl) [reply manifests/init.pp:77]: Then I'd change the data type. I also wonder why it's a `Numeric`. Isn't a pin always an integer? https://manpages.debian.org/buster/apt/apt_preferences.5.en.html#How_APT_Interprets_Priorities suggests it is.
```suggestion
  Variant[Integer, String] $package_apt_pin = 200,
```

**Review by nmaludy — APPROVED** [review]
Did some research as well and can confirm that the erlang repos now recommend setting themselves as higher priority.

File: manifests/init.pp (hunk 1 above)
Comment (wyardley) [reply manifests/init.pp:77]: That was my thought as well (changing data type), but are there use cases where someone would want / need to override it to being unset? I can just adjust the docs and make it non-optional as long as everyone’s good with that.

Comment (kenyon) [reply manifests/init.pp:77]: It can still be unset by setting it to `undef`.

Comment (kenyon) [reply manifests/init.pp:77]: The data types would also need to be changed in `repo/apt/bintray.pp` and `repo/apt/erlang_solutions.pp`.

**Comment by kenyon** [comment]
> Did some research as well and can confirm that the erlang repos now recommend setting themselves as higher priority.

This PR is doing the opposite, deprioritizing the non-debian.org repos (the default priority is 500).

File: manifests/init.pp (hunk 1 above)
Comment (wyardley) [reply manifests/init.pp:77]: @kenyon at one point, you could not set a default value for an optional parameter. I'm not sure if that's still the case, though

Comment (kenyon) [reply manifests/init.pp:77]: @wyardley my comment was under the assumption you implement @ekohl's suggestion, so it wouldn't be optional.

**Comment by wyardley** [comment]
> This PR is doing the opposite, deprioritizing the non-debian.org repos (the default priority is 500).

You're right, which is not what I was intending. I'll switch the priority to 600.

However, I think this is causing the same issue as before where there are conflicts between the vendor provided ones and the ones from erlang-solutions.

```
  The following packages have unmet dependencies:
   erlang : Depends: erlang-typer
  E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
```

Now that I've added a test for package source in the integration test, it's clear that in a lot of cases, the package that's getting installed is not the expected one.

**Comment by wyardley** [comment]
Yeah, this seems to not be the actual issue with ensuring absent and / or with the integration test failures.

**Pull request closed without merging**